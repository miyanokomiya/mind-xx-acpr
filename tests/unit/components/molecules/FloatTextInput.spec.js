import { mount, shallowMount } from '@vue/test-utils'
import Target from '@/components/molecules/FloatTextInput.vue'

describe('components/molecules/FloatTextInput.vue', () => {
  describe('snapshot', () => {
    it('default', () => {
      const wrapper = mount(Target, {
        propsData: {
          value: 'text',
          targetKey: 'target',
          x: 10,
          y: 20,
        },
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('computed', () => {
    describe('localValue', () => {
      it('set => $emit', () => {
        const wrapper = shallowMount(Target, {
          propsData: {
            value: 'text',
            targetKey: 'target',
            x: 10,
            y: 20,
          },
        })
        wrapper.vm.localValue = 'updated'
        expect(wrapper.emitted('input')).toEqual([['updated']])
      })
    })
  })

  describe('methods', () => {
    describe('done', () => {
      it('$emit', () => {
        const wrapper = shallowMount(Target, {
          propsData: {
            value: 'text',
            targetKey: 'target',
            x: 10,
            y: 20,
          },
        })
        wrapper.vm.done()
        expect(wrapper.emitted('done')).toEqual([
          [
            {
              value: 'text',
              targetKey: 'target',
            },
          ],
        ])
      })
    })
  })
})
