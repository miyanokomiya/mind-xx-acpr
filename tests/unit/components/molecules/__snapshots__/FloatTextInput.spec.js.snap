// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/FloatTextInput.vue snapshot default 1`] = `
<div
  class="float-text-input-wrapper"
  style="position: absolute; left: 0px; top: 0px; transform: translate(10px, 20px);"
>
  <div
    class="v-input text-field v-textarea v-textarea--no-resize v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-text-field__slot"
        >
          <textarea
            autofocus="autofocus"
            id="input-2"
            rows="3"
          />
        </div>
      </div>
    </div>
  </div>
   
  <button
    class="submit v-btn v-btn--contained v-btn--fab v-btn--round theme--dark v-size--small primary"
    type="button"
  >
    <span
      class="v-btn__content"
    >
      <i
        aria-hidden="true"
        class="v-icon notranslate material-icons theme--dark"
        style="line-height: 32px;"
      >
        done
      </i>
    </span>
  </button>
</div>
`;
